### PR TITLE
[5.3] Added support for notifications MailChannel using messages that implements Mailable contract

### PR DESCRIPTION
**Pull request by butschster**
``` php
...

public function toMail($notifiable)
{
      return (new \App\Mail\OrderShipped($this->order))->to($notifiable->routeNotificationFor('mail'));
}
...
```

### CODE REVIEW

**Comment by themsaid** [comment]
Can you please add some tests?

File: tests/Notifications/NotificationMailChannelTest.php [hunk 1]
```diff
@@ -272,3 +286,23 @@ public function toMail($notifiable)
...
 }
+
+class NotificationMailChannelTestNotificationWithMailableContract extends Notification
+{
+    public function toMail($notifiable)
+    {
+        $mock = Mockery::mock(Illuminate\Contracts\Mail\Mailable::class);
+
+        $mock->shouldReceive('send')->once()->with(Mockery::on(function($mailer) {
+            if (! ($mailer instanceof \Illuminate\Contracts\Mail\Mailer)) {
```
Comment (GrahamCampbell) [inline tests/Notifications/NotificationMailChannelTest.php]: No leading slash please. Also, no extra brackets are needed.

File: tests/Notifications/NotificationMailChannelTest.php [hunk 2]
```diff
@@ -210,6 +210,20 @@ public function testMessageWithToAddress()
...
+    public function testMessageWithMailableContract()
+    {
+        $notification = new NotificationMailChannelTestNotificationWithMailableContract;
+        $notifiable = new NotificationMailChannelTestNotifiable;
+
+        $channel = new Illuminate\Notifications\Channels\MailChannel(
+            $mailer = Mockery::mock(Illuminate\Contracts\Mail\Mailer::class)
+        );
+
+        $mailer->shouldReceive('send')->once();
```
Comment (themsaid) [inline tests/Notifications/NotificationMailChannelTest.php]: Mailer would receive send anyway, I think you need to test that the Mailable object would receive send in this case.

Comment (themsaid) [inline tests/Notifications/NotificationMailChannelTest.php]: Sorry, didn't notice the test below.

**Comment by tillkruss** [comment]
@butschster It would be nice if that would be in the docs, feel like submitting a PR?

**Pull request merged**